code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm

```diff
@@ -50,6 +50,13 @@
 				if(5 to INFINITY)
 					user << "<span class='info'>Such a feast! [target] will yield much essence to you.</span>"
 			sleep(30)
+			if(!in_range(user, target))
```

Comment:
Might want to check this; if you move and break the range, I believe this makes it so you can't absorb anyone else, permanently as it thinks you're still absorbing.